dev-docs/references/directory_layout.md

```diff
@@ -0,0 +1,20 @@
+# Directory layout
```

Comment:
I think not directly, since we also describe folders that are not totally related to UA code architecture. So I think this is a broader concept. But let me know if you disagree

Reply:
I think this should really be broken into two reference documents.

1. directory layout of the source/git repo
2. layout of files that get installed by the deb package

But this PR is already large so we don't have to do that now. I think it is fine as is for now.